xknx/dpt/dpt_1byte_control.py

```diff
@@ -0,0 +1,259 @@
+        #   1: + = increase/move down
+        control = 0 if value <= 0 else 1
+
+        stepcode = (
+            0  # special case = break indication (e.g. stop dimming/moving blinds)
+        )
+        if abs(value) >= 100:
+            stepcode = 1
+        elif abs(value) >= 50:
+            stepcode = 2
+        elif abs(value) >= 25:
+            stepcode = 3
+        elif abs(value) >= 12:
+            stepcode = 4
+        elif abs(value) >= 6:
+            stepcode = 5
+        elif abs(value) >= 3:
+            stepcode = 6
+        elif abs(value) >= 1:
+            stepcode = 7
+
+        return {"control": control, "step_code": stepcode}
+
+    @staticmethod
+    def _to_increment(value):
+        """Calculate the increment value from the stepcode and control bit as defined in the KNX standard section 3.3.1."""
+        # calculated using floor(100/2^((value&0x07)-1))
+        inc = [0, 100, 50, 25, 12, 6, 3, 1][value["step_code"] & 0x07]
+        return inc if value["control"] == 1 else -inc
+
+    @classmethod
+    def to_knx(cls, value, invert=False):
+        """Serialize to KNX/IP raw data."""
+        if not isinstance(value, int):
+            raise ConversionError("Cant serialize %s" % cls.__name__, value=value)
+
+        return super().to_knx(cls._from_increment(value), invert)
+
+    @classmethod
+    def from_knx(cls, raw, invert=False):
+        """Parse/deserialize from KNX/IP raw data."""
+        return cls._to_increment(super().from_knx(raw, invert))
+
+
+class DPTControlStepwiseDimming(DPTControlStepwise):
+    """Abstraction for KNX DPT 3.007 / DPT_Control_Dimming in stepwise mode."""
+
+    dpt_main_number = 3
+    dpt_sub_number = 7
+    value_type = "stepwise_dimming"
+
+
+class DPTControlStepwiseBlinds(DPTControlStepwise):
+    """Abstraction for KNX DPT 3.008 / DPT_Control_Blinds in stepwise mode."""
+
+    dpt_main_number = 3
+    dpt_sub_number = 8
+    value_type = "stepwise_blinds"
+
+
+class TitleEnum(Enum):
+    """Enum with a descriptive string representation.
+
+    Ensures values are rendered nicely, e.g. in home assistant.
+    """
+
+    def __str__(self):
+        """Return string representation."""
+        # pylint: disable=no-member
+        return self.name.title()
+
+
+class DPTControlStartStop(DPTControl):
+    """Abstraction for KNX DPT 3.xxx in start/stop mode."""
+
+    value_type = "startstop"
+    unit = ""
+
+    class Direction(TitleEnum):
+        """Enum for indicating the direction."""
+
+        DECREASE = 0
+        INCREASE = 1
+        STOP = 2
+
+    @classmethod
+    def to_knx(cls, value, invert=False):
+        """Convert value to payload."""
+        control = 0
+        step_code = 0
+        if value == cls.Direction(1):  # INCREASE/DOWN
+            control = 1
+            step_code = 1
+        elif value == cls.Direction(0):  # DECREASE/UP
+            control = 0
+            step_code = 1
+        elif value == cls.Direction(2):  # STOP
```

Comment:
```suggestion
        if value == cls.Direction.INCREASE:  # Down
            control = 1
            step_code = 1
        elif value == cls.Direction.DECREASE:  # Up
            control = 0
            step_code = 1
        elif value == cls.Direction.STOP:
```

Reply:
Nop, not possible. I admit the Enum implementation is a little bit tricky. The idea is to use the 0 / 1 representation in the base class and thus for the encoding. And only in the subclasses I provide the interpretation, whether 0 is now DECREASE or UP.
It is necessary to really use the numeric represenation here. As the concrete interpretation of Direction is overridden when called from a derived class. Consider the following example

```
c = DPTControlStartStopBlinds()
v=c.from_knx(1)
print(v) # => Up
print(c.to_knx(v)) => 1
```

When I follow your suggestion I would get the following exception in the last line

```
Traceback (most recent call last):
  File "test.py", line 242, in <module>
    print(c.to_knx(v))
  File "test.py", line 186, in to_knx
    if value == cls.Direction.INCREASE:  # INCREASE/DOWN
  File "C:\tools\miniconda3\lib\enum.py", line 346, in __getattr__
    raise AttributeError(name) from None
AttributeError: INCREASE
```

`INCREASE` is not defined, as `DPTControlStartStopBlinds`'s `Direction` only knows `UP`, `DOWN` and `STOP`.